app/assets/javascripts/oxalis/controller/combinations/skeletontracing_plane_controller.js

```diff
@@ -43,179 +43,173 @@ const OrthoViewToNumber: OrthoViewMapType<number> = {
+
+  // prevent flickering sometimes caused by picking
+  planeView.renderFunction(true);
+
+  const skeletonTracing = enforceSkeletonTracing(Store.getState().tracing);
+  // otherwise we have hit the background and do nothing
+  if (nodeId > 0) {
+    if (altPressed) {
+      getActiveNode(skeletonTracing).map(activeNode =>
+        Store.dispatch(mergeTreesAction(activeNode.id, nodeId)),
+      );
+    } else if (ctrlPressed) {
+      getActiveNode(skeletonTracing).map(activeNode =>
+        Store.dispatch(deleteEdgeAction(activeNode.id, nodeId)),
+      );
+    } else {
+      Store.dispatch(setActiveNodeAction(nodeId));
     }
-    const activeNodeMaybe = getActiveNode(Store.getState().tracing);
-
-    // set the new trace direction
-    activeNodeMaybe.map(activeNode =>
-      Store.dispatch(
-        setDirectionAction([
-          position[0] - activeNode.position[0],
-          position[1] - activeNode.position[1],
-          position[2] - activeNode.position[2],
-        ]),
-      ),
-    );
+  }
+}
 
-    const rotation = getRotationOrtho(activeViewport);
-    this.addNode(position, rotation, !ctrlPressed);
+function setWaypoint(position: Vector3, ctrlPressed: boolean): void {
+  const { activeViewport } = Store.getState().viewModeData.plane;
+  if (activeViewport === OrthoViews.TDView) {
+    return;
+  }
+  const skeletonTracing = enforceSkeletonTracing(Store.getState().tracing);
+  const activeNodeMaybe = getActiveNode(skeletonTracing);
 
-    // Strg + Rightclick to set new not active branchpoint
-    const newNodeNewTree = Store.getState().userConfiguration.newNodeNewTree;
-    if (ctrlPressed && !newNodeNewTree) {
-      Store.dispatch(createBranchPointAction());
-      activeNodeMaybe.map(activeNode => Store.dispatch(setActiveNodeAction(activeNode.id)));
-    }
+  // set the new trace direction
+  activeNodeMaybe.map(activeNode =>
+    Store.dispatch(
+      setDirectionAction([
+        position[0] - activeNode.position[0],
+        position[1] - activeNode.position[1],
+        position[2] - activeNode.position[2],
+      ]),
+    ),
+  );
+
+  const rotation = getRotationOrtho(activeViewport);
+  addNode(position, rotation, !ctrlPressed);
+
+  // Strg + Rightclick to set new not active branchpoint
+  const { newNodeNewTree } = Store.getState().userConfiguration;
+  if (ctrlPressed && !newNodeNewTree) {
+    Store.dispatch(createBranchPointAction());
+    activeNodeMaybe.map(activeNode => Store.dispatch(setActiveNodeAction(activeNode.id)));
   }
+}
 
-  addNode = (position: Vector3, rotation: Vector3, centered: boolean): void => {
-    const state = Store.getState();
-    const { newNodeNewTree } = state.userConfiguration;
-    const activeNodeMaybe = getActiveNode(state.tracing);
+function addNode(position: Vector3, rotation: Vector3, centered: boolean): void {
+  const state = Store.getState();
+  const { newNodeNewTree } = state.userConfiguration;
+  const activeNodeMaybe = enforce(getActiveNode)(state.tracing.skeleton);
```

Comment:
No, there isn't a big difference. `enforceSkeletonTracing` can make sense, if you want to do various things with the tracing, since you only enforce its existence once. `enforce` is a nice wrapper in my opinion, if you are going to use an accessor anyway. In general, both approaches are fine, though.